crates/holochain/src/core/workflow/app_validation_workflow.rs

```diff
@@ -1,4 +1,80 @@
-//! The workflow and queue consumer for sys validation
+//! Holochain workflow to validate all incoming DHT operations with an
+//! app-defined validation function.
+//!
+//! Triggered by system validation, this workflow iterates over a list of
+//! [`DhtOp`]s to be validated, validates each op, updates its validation status
+//! in the database accordingly, and triggers op integration if necessary.
+//!
+//! # Sequential validation
+//!
+//! Ops are validated in sequence based on their op type and the timestamp they
+//! were authored (see [`OpOrder`] and [`OpNumericalOrder`]). Validating one op
+//! after the other with this ordering ensures that ops that depend on earlier
+//! ops will be validated after the earlier ops. An example is an incoming delete
+//! op that depends on a create op. Validated in order of their authoring, the
+//! create op is validated first, followed at some stage by the delete op. If
+//! the validation function references the original action when validating
+//! delete ops, the create op will have been validated and is available in the
+//! database. Otherwise the delete op could not be validated and its dependency,
+//! the create op, would be awaited first.
+//!
+
+// This does not apply right now, because all actions are written to the database
+// straight away in the incoming dht ops workflow. see https://github.com/holochain/holochain/issues/3724
+
+//! # Op validation
+//!
+//! For each op the [corresponding app validation function](https://docs.rs/hdi/latest/hdi/#data-validation)
+//! is executed. Entries and links, that the ops are derived from, have been
```

Comment:
```suggestion
//! is executed. Entry and link CRUD actions, which the ops are derived from, have been
```
I know this throws off the character count for the line, sorry

Reply:
Actually, this brings up another question: what integrity zome(s) validate(s) ops produced by a `Delete` or `DeleteLink`? This question comes up further down too.

Reply:
> I know this throws off the character count for the line, sorry

An, no worries. I don't even know why I'm staying within 80 chars, it's inconsistent in the repo.

> what integrity zome(s) validate(s) ops produced by a Delete or DeleteLink?

It's the integrity zome that contains the entry and link types that were used to create them. But it must be the same integrity zome for all CRUD actions anyway, you cannot have an entry or link type across multiple zomes.